### PR TITLE
convert all f32 things to f64 things

### PR DESCRIPTION
**Pull request by trusktr**
@corruptedzulu I made a new branch off yours. I replaced all f64s with f32s.

Some tests are still failing as before, but this time no more type casting which may save CPU. Note the `Mathf` usage in place of `Math`. It is the same, but with all f32 operations instead of f64.

I guess Prettier somehow formatted files that the find/replacing I did didn't touch. Setting the "Hide whitespace changes" option in the "Files changed" tab (in the gear cog at top left of modified files) may make it a little easier to read.

Let's get your branch passing and merged first, then we can rebase and merge this one.

### CODE REVIEW

**Comment by corruptedzulu** [comment]
I sent in two additional commits for this to correct some issues. @trusktr can you review those commits and give a yea/nay on them? I'll then mark the whole PR as reviewed and merge it into the geometry branch

File: src/as/math/Math.ts [hunk 1]
```diff
@@ -54,10 +53,11 @@ export function generateUUID(): string {
...
 	NativeMathf.seedRandom(12345678)
-	var d0: i32 = i32(Mathf.random() * 0xffffffff) //| 0
-	var d1: i32 = i32(Mathf.random() * 0xffffffff) //| 0
-	var d2: i32 = i32(Mathf.random() * 0xffffffff) //| 0
-	var d3: i32 = i32(Mathf.random() * 0xffffffff) //| 0
+
+	var d0: u32 = u32(Mathf.random() * 0xffffffff) //| 0
+	var d1: u32 = u32(Mathf.random() * 0xffffffff) //| 0
+	var d2: u32 = u32(Mathf.random() * 0xffffffff) //| 0
+	var d3: u32 = u32(Mathf.random() * 0xffffffff) //| 0
```
Comment (trusktr) [inline src/as/math/Math.ts:60]: I think `|0` was a cast to int in plain JS. F.e., running `5.2356 | 0` in your browser console should show `5`.

File: src/as/core/Geometry.spec.ts [hunk 2]
```diff
@@ -136,17 +136,17 @@ describe('Geometry', () => {
...
 		// assert(v1.x === -0.5 && v1.y === 0 && v1.z < Number.EPSILON, 'second vertex was rotated')
-		//expect<bool>(v1.x === -0.5 && v1.y === 0 && v1.z < Number.EPSILON).toBe(true)
+		expect<bool>(v1.x === -0.5 && v1.y === 0 && v1.z < f32.EPSILON).toBe(true)
 		// expect<bool>(v1.x === -0.5 && v1.y === 0 && v1.z < Mathf.pow(2, -52)).toBe(true, 'v1 values passed')
-		expect(v1.x === -0.5).toBe(true, 'v1 x')
-		expect(v1.y === 0).toBe(true, 'v1 y')
-		//expect(v1.z < f32(Mathf.pow(2, -52))).toBe(true, 'v1 z')
+		// expect(v1.x === -0.5).toBe(true, 'v1 x')
+		// expect(v1.y === 0).toBe(true, 'v1 y')
+		// expect(v1.z < f32(Mathf.pow(2, -52))).toBe(true, 'v1 z')
```
Comment (trusktr) [inline src/as/core/Geometry.spec.ts:148]: Just curious, did you leave some of them commented, but planning to work on once we merge this one?

Comment (trusktr) [reply src/as/core/Geometry.spec.ts:148]: Looks like the `|0` just floors the value. Is that was the `u32()` cast does too?

File: src/as/math/Math.ts (hunk 1 above)
Comment (trusktr) [reply src/as/math/Math.ts:60]: Looks like the `|0` just floors the value. Is that was the `u32()` cast does too?

Comment (jtenner) [reply src/as/math/Math.ts:60]: That is correct. Yes.

File: src/as/core/Geometry.spec.ts (hunk 2 above)
Comment (jtenner) [reply src/as/core/Geometry.spec.ts:148]: Why isn't this expect(v1.x).toBe(-0.5)?

File: src/as/math/Math.ts (hunk 1 above)
Comment (trusktr) [reply src/as/math/Math.ts:60]: @jtenner That original code was like this.

https://github.com/mrdoob/three.js/blob/dev/src/math/MathUtils.js#L25-L28

Do you recommend any other way to do it?

File: src/as/core/Geometry.spec.ts (hunk 2 above)
Comment (trusktr) [reply src/as/core/Geometry.spec.ts:148]: It will be!

**Pull request merged**